File: src/HolzShots/Input/Actions/OpenSettingsJsonCommand.vb
Imports System.Threading.Tasks
Imports HolzShots.Composition.Command

Namespace Input.Actions
    <Command("openSettingsJson")>
    Public Class OpenSettingsJsonCommand
        Implements ICommand(Of HSSettings)

        Public Function Invoke(parameters As IReadOnlyDictionary(Of String, String), settingsContext As HSSettings) As Task Implements ICommand(Of HSSettings).Invoke
            If parameters Is Nothing Then Throw New ArgumentNullException(NameOf(parameters))
            If settingsContext Is Nothing Then Throw New ArgumentNullException(NameOf(settingsContext))

            UserSettings.OpenSettingsInDefaultEditor()
            Return Task.CompletedTask
        End Function
    End Class
End Namespace

File: src/HolzShots/Input/Action/OpenSettingsJsonCommand.cs
using System.Collections.Generic;
using HolzShots.Composition.Command;

namespace HolzShots.Input.Actions;

[Command("openSettingsJson")]
public class OpenSettingsJsonCommand : ICommand<HSSettings>
{
    public Task Invoke(IReadOnlyDictionary<string, string> parameters, HSSettings settingsContext)
    {
        ArgumentNullException.ThrowIfNull(parameters);
        ArgumentNullException.ThrowIfNull(settingsContext);

        UserSettings.OpenSettingsInDefaultEditor();
        return Task.CompletedTask;
    }
}
